compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -209,10 +209,8 @@ fn resolve_associated_item<'tcx>(
                 let name = tcx.item_name(trait_item_id);
                 if name == sym::clone {
                     let self_ty = trait_ref.self_ty();
-
-                    let is_copy = self_ty.is_copy_modulo_regions(tcx, param_env);
```

Comment:
that one is only for consistency, is it? Here the `param_env` is actually correct

Reply:
Whoops, forgot to respond. Yes, the param-env here is correct, it's just not needed. We would've had to pass down `is_copy` rather than recomputing it when buidling the shim, but it's easier just to make the tykind match below exhaustive.